### PR TITLE
[docs] update documentation for integration user-authn with Gitlab ver16+

### PR DESCRIPTION
**Pull request by ishtarov**
## Description
A new option has been added to the GitLab integration setup in version 16; users are faced with difficulties during authorization if it is not configured. The documentation has been updated with a method for enabling this option.

```changes
section: docs
type: chore
summary: Update documentation for integration user-authn with Gitlab ver16+.
impact_level: low
```

### CODE REVIEW

File: modules/150-user-authn/docs/USAGE_RU.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ spec:
 Для этого необходимо:
 * **self-hosted**: перейти в `Admin area` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`;
 * **cloud gitlab.com**: под главной учетной записью проекта перейти в `User Settings` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`.
+* При использовании Gitlab версии 16 и выше - при создании приложения также требуется включить опцию `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow`.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE_RU.md:77]:
```suggestion
* При использовании GitLab версии 16 и выше, при создании приложения также требуется включить опцию `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow`.
```

File: modules/150-user-authn/docs/USAGE.md [hunk 2]
```diff
@@ -74,6 +74,7 @@ Create a new application in the GitLab project.
 To do this, you need to:
 * **self-hosted**: go to `Admin area` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)` and set scopes `read_user`, `openid`;
 * **cloud gitlab.com**: under the main project account, go to `User Settings` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)`; also, don't forget to set scopes `read_user`, `openid`.
+* When using Gitlab version 16 and higher, when creating an application, you also need to enable the `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow` checkbox.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE.md:77]:
```suggestion
* When using GitLab version 16 and higher, when creating an application, you also need to enable the `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow` checkbox.
```

File: modules/150-user-authn/docs/USAGE.md [hunk 3]
```diff
@@ -74,6 +74,7 @@ Create a new application in the GitLab project.
 To do this, you need to:
 * **self-hosted**: go to `Admin area` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)` and set scopes `read_user`, `openid`;
 * **cloud gitlab.com**: under the main project account, go to `User Settings` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)`; also, don't forget to set scopes `read_user`, `openid`.
+* When using GitLab version 16 and higher, when creating an application, you also need to enable the `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow` checkbox.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE.md:77]:
```suggestion
* (For GitLab version starting with 16) Enable the `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow` checkbox  when creating an application.
```

File: modules/150-user-authn/docs/USAGE_RU.md [hunk 4]
```diff
@@ -74,6 +74,7 @@ spec:
 Для этого необходимо:
 * **self-hosted**: перейти в `Admin area` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`;
 * **cloud gitlab.com**: под главной учетной записью проекта перейти в `User Settings` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`.
+* При использовании GitLab версии 16 и выше, при создании приложения также требуется включить опцию `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow`.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE_RU.md:77]:
```suggestion
* (для GitLab версии 16 и выше) включить опцию `Trusted`/`Trusted applications are automatically authorized on Gitlab OAuth flow` при создании приложения.
```

File: modules/150-user-authn/docs/USAGE.md [hunk 5]
```diff
@@ -74,6 +74,7 @@ Create a new application in the GitLab project.
 To do this, you need to:
 * **self-hosted**: go to `Admin area` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)` and set scopes `read_user`, `openid`;
 * **cloud gitlab.com**: under the main project account, go to `User Settings` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)`; also, don't forget to set scopes `read_user`, `openid`.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE.md:76]:
```suggestion
* **cloud gitlab.com**: under the main project account, go to `User Settings` -> `Application` -> `New application` and specify the `https://dex.<modules.publicDomainTemplate>/callback` address as the `Redirect URI (Callback url)`; also, don't forget to set scopes `read_user`, `openid`;
```

File: modules/150-user-authn/docs/USAGE_RU.md [hunk 6]
```diff
@@ -74,6 +74,7 @@ spec:
 Для этого необходимо:
 * **self-hosted**: перейти в `Admin area` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`;
 * **cloud gitlab.com**: под главной учетной записью проекта перейти в `User Settings` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`.
```
Comment (z9r5) [inline modules/150-user-authn/docs/USAGE_RU.md:76]:
```suggestion
* **cloud gitlab.com**: под главной учетной записью проекта перейти в `User Settings` -> `Application` -> `New application` и в качестве `Redirect URI (Callback url)` указать адрес `https://dex.<modules.publicDomainTemplate>/callback`, scopes выбрать: `read_user`, `openid`;
```

**Comment by z9r5** [comment]
@ishtarov Can you fix the DCO test, please? Thanks.

**Review by nabokihms — APPROVED** [review]

**Pull request merged**